src/main/kotlin/uk/gov/justice/digital/hmpps/supportadditionalneedsapi/service/ReviewScheduleService.kt

```diff
@@ -51,4 +54,36 @@ class ReviewScheduleService(
     reviewScheduleRepository.save(reviewScheduleEntity)
     eventPublisher.createAndPublishReviewScheduleEvent(prisonNumber)
   }
+
+  fun createOrUpdate(prisonNumber: String, startDate: LocalDate, fundingType: String) {
+    val existing = reviewScheduleRepository.findFirstByPrisonNumberOrderByUpdatedAtDesc(prisonNumber)
+    val proposedDeadline = getDeadlineDate(startDate)
+
+    // No SCHEDULED review schedule - create a fresh one with the new deadline
+    if (existing == null || existing.status != ReviewScheduleStatus.SCHEDULED) {
+      createReviewSchedule(
+        prisonNumber = prisonNumber,
+        reviewDate = proposedDeadline,
+        prisonId = "N/A",
```

Comment:
Prison ID ? Will that always be N/A ? or is that only as part of this initial impl, and this will get fleshed out as we go along?